File: algorithms/L/Largest 3-Same-Digit Number in String/Largest 3-Same-Digit Number in String.js
var largestGoodInteger = function(num) {
    let maxGoodInt = '';
    for (let i = 0; i <= num.length - 3; i++) {
        if (num[i] === num[i+1] && num[i+1] === num[i+2]) {
            if (num[i] >= maxGoodInt) {
                maxGoodInt = num[i];
            }
        }
    }
    return maxGoodInt + maxGoodInt + maxGoodInt;
};

File: algorithms/L/Largest 3-Same-Digit Number in String/Largest 3-Same-Digit Number in String.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string ans = "";
        for(int i=1; i<num.size()-1; i++) {
            if(num[i-1] == num[i] && num[i] == num[i+1]) {
				string temp = {num[i-1], num[i], num[i+1]};
                ans = max(ans, temp);
            }
        }
        return ans;    
    }
};
